File: 02-programacion-basica/03-tipos-de-datos-y-conversiones/java/ejemplos-guiados/TiposYConversiones.java
public class TiposYConversiones {
    public static void main(String[] args) {
        String textoEdad = "21";

        // parseInt transforma texto numérico en int. Puede fallar si el texto no representa un entero.
        int edad = Integer.parseInt(textoEdad);
        double promedio = 17.75;

        // El casting a int descarta la parte decimal; no redondea.
        int parteEntera = (int) promedio;
        boolean mayorEdad = edad >= 18;

        System.out.println("Edad: " + edad);
        System.out.println("Parte entera del promedio: " + parteEntera);
        System.out.println("¿Es mayor de edad?: " + mayorEdad);
    }
}

File: 02-programacion-basica/03-tipos-de-datos-y-conversiones/scala/ejemplos-guiados/TiposYConversiones.scala
object TiposYConversiones:
  def main(args: Array[String]): Unit =
    val textoEdad: String = "21"
    val edad: Int = textoEdad.toInt
    val promedio: Double = 17.75

    // toInt descarta la parte decimal; no realiza redondeo.
    val parteEntera: Int = promedio.toInt
    val mayorEdad: Boolean = edad >= 18

    println(s"Edad: $edad")
    println(s"Parte entera del promedio: $parteEntera")
    println(s"¿Es mayor de edad?: $mayorEdad")
